### PR TITLE
Bug fix for CLI feature

### PR DESCRIPTION
**Pull request by hsubramanianaks**
This PR is for fixing https://github.com/Azure/Bridge-To-Kubernetes/issues/81

1. Now script writes into /usr/local/bin/bridgetokubernetes and symlinks dsc -->  /usr/local/bin/bridgetokubernetes. 
2. same for $HOME/.local/bin/bridgetokubernetes/ folder. 

Thank you for @teejaded and @Tatsinnit for cool suggestions. 

https://raw.githubusercontent.com/hsubramanianaks/Bridge-To-Kubernetes/bug/fix-install-script/scripts/install.sh

Testing screenshots

**WSL (ubuntu): without running source ~.profile**
image

image

WSL (ubuntu) with running source ~.profile

image

image

GitBash from windows:

image

for git bash symlink doesn't work for now, windows user has to cd into the installation directory to use dsc 

image

Symblink:
image

### CODE REVIEW

**Comment by harans87** [comment]
Worked in Mac as well 

Screen Shot 2022-10-27 at 10 58 54 PM

**Comment by teejaded** [comment]
It works well to install, but if you run it a second time, like you might if you were upgrading, it errors out.

```
Fri Oct 28 05:12:50 UTC 2022 - WARNING - installation target directory is write protected, run as root to override
Fri Oct 28 05:12:50 UTC 2022 - INFO - removing directory:/usr/local/bin/bridgetokubernetes
rm: /usr/local/bin/bridgetokubernetes/Microsoft.Rest.ClientRuntime.Azure.Authentication.dll: Permission denied
rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Cryptography.Internal.dll: Permission denied
rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Connections.Abstractions.dll: Permission denied
...
rm: /usr/local/bin/bridgetokubernetes: Permission denied
```

**Comment by hsubramanianaks** [comment]
> It works well to install, but if you run it a second time, like you might if you were upgrading, it errors out.
> 
> ```
> Fri Oct 28 05:12:50 UTC 2022 - WARNING - installation target directory is write protected, run as root to override
> Fri Oct 28 05:12:50 UTC 2022 - INFO - removing directory:/usr/local/bin/bridgetokubernetes
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.Rest.ClientRuntime.Azure.Authentication.dll: Permission denied
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Cryptography.Internal.dll: Permission denied
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Connections.Abstractions.dll: Permission denied
> ...
> rm: /usr/local/bin/bridgetokubernetes: Permission denied
> ```



> It works well to install, but if you run it a second time, like you might if you were upgrading, it errors out.
> 
> ```
> Fri Oct 28 05:12:50 UTC 2022 - WARNING - installation target directory is write protected, run as root to override
> Fri Oct 28 05:12:50 UTC 2022 - INFO - removing directory:/usr/local/bin/bridgetokubernetes
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.Rest.ClientRuntime.Azure.Authentication.dll: Permission denied
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Cryptography.Internal.dll: Permission denied
> rm: /usr/local/bin/bridgetokubernetes/Microsoft.AspNetCore.Connections.Abstractions.dll: Permission denied
> ...
> rm: /usr/local/bin/bridgetokubernetes: Permission denied
> ```

@teejaded  Please check now.

File: .github/workflows/integration-check-install.yml [hunk 1]
```diff
@@ -5,17 +5,26 @@ on:
 
 jobs:
   install:
-    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```
Comment (elenavillamil) [inline .github/workflows/integration-check-install.yml:10]: Should we have same checks for windows?

File: scripts/install.sh [hunk 2]
```diff
@@ -193,24 +193,36 @@ copy_b2k_files() {
...
 remove_tmp_dirs() {
-    log INFO "removing temp directory:$1"
+    log INFO "removing directory:$1"
     if [ -d $1 ]; then
-        rm -rf $1
+        $2 rm -rf $1
+    fi
+}
+
+create_sym_link() {
```
Comment (elenavillamil) [inline scripts/install.sh:220]: Can we add comment on background of why symlink is needed. just by looking at command ln -sf $1 $2 is very hard for reader to know what we are trying to do and why.

File: .github/workflows/integration-check-install.yml (hunk 1 above)
Comment (hsubramanianaks) [reply .github/workflows/integration-check-install.yml:10]: Planned to add them , windows with gitbash for now.

**Comment by teejaded** [comment]
lgtm

**Review by jaiveerk — APPROVED** [review]

File: scripts/install.sh [hunk 3]
```diff
@@ -193,24 +193,37 @@ copy_b2k_files() {
...
     cd ~
     remove_tmp_dirs $HOME/tmp/bridgetokubernetes
 }
 
 remove_tmp_dirs() {
-    log INFO "removing temp directory:$1"
+    log INFO "removing directory:$1"
     if [ -d $1 ]; then
-        rm -rf $1
+        $2 rm -rf $1
```
Comment (Tatsinnit) [inline scripts/install.sh:216]: 💡thank you for this and addition in review: what is the probable value of “$2” here?

**Review by Tatsinnit — APPROVED** [review]

File: scripts/install.sh (hunk 3 above)
Comment (hsubramanianaks) [reply scripts/install.sh:216]: $2 = Sudo or empty :)

**Pull request merged**